### ISSUE TITLE
Prisoner numbering rule was not clear.

### ISSUE THREAD

**Issue by tomohisaota** [tarao/prisoners-switch#13]
The knowledge of "numbering rule" is important.
If the number is random, I don't know good way to solve this problem.

I think it worth mentioning in source code.
Each prisoner has unique sequential number (0 to rule.TotalPrisoners-1)

**Comment by tomohisaota** [comment]
Actually, that regulation seem to be too tight.
With 0-99 sequence, I can safely assign special role to prisoner with number 0.
With random (but unique and consistent) number, it is really hard to assign some role to specific prisoner.

Prisoners can talk to each other before starting this mission.
So I think it is fair to use sequence rather than random number.

**Comment by tomohisaota** [comment]
Make pull req
https://github.com/tarao/prisoners-switch/pull/14

Thank you for creating interesting quiz. Pull req is least I can do :-)

**Comment by tarao** [comment]
Merged.

**Comment by tarao** [comment]
I didn't commeted about the sequetial numbers.

> Prisoners can talk to each other before starting this mission.
> So I think it is fair to use sequence rather than random number.

Sounds reasonable!

> it is really hard to assign some role to specific prisoner

If you really have to do that, you will, for example, record the very first number throuout the different runs of games to remember who is the counter.  Perhaps you need a lock for accessing the recorded value because it is racy, and this is indeed quite hard wihout `import`ing `sync` package. It is possible by using channels but such a Golang hack is out of scope the problem.

**Comment by tomohisaota** [comment]
Thanks to go channel, It is actually quite easy to remember the very first prisoner's number, and choose him as counter.
https://github.com/tomohisaota/prisoners-switch/blob/tomohisaota-cheat/strategy/my_strategy.go
{"success":true,"message":"All games passed","steps":1096105,"used_switches":1,"score":189038}

I have confirmed that it works.
And it does not have any additional switching nor steps.
But I think sharing information through channel is cheating :-p
